### ISSUE TITLE
Import Manual v1.0 (English version / 英語版)

### ISSUE THREAD

**Issue by mapconcierge** [yuuhayashi/citygml-osm#97]
ここに貼り付けよろしく！  息吹

**Comment by ibuki** [comment]
@mapconcierge 
初めまして。人違いだと思いますので、メンションご確認ください。

**Comment by mapconcierge** [comment]
@ibuki さん、失礼しました！！ 間違えました！

**Comment by ibuki76** [comment]
# Plateau building data: Import procedure into OpenStreetMap (Draft)
# English Ver.

* This section describes the procedure for importing 3D data building objects published by the Plateau project into OpenStreetMap.
* Please refer to [the OSM wiki: JA:MLIT PLATEAU/imports outline](https://wiki.openstreetmap.org/wiki/JA:MLIT_PLATEAU/imports_outline) for information on the entire import process.
* Please make sure you have the latest versions of [JAVA](https://www.java.com/ja/) and [JOSM](https://josm.openstreetmap.de/) installed on your PC before starting the following process.

## 1. Download the Citygml file of the target area.
Download the file to be worked on from the [G-Spatial Information Center](https://www.geospatial.jp/ckan/dataset/plateau) .

* [3D city model (Project PLATEAU) portal site](https://www.geospatial.jp/ckan/dataset/plateau)
* When downloading, select the `CityGML` format.

After downloading the file, extract the zip file.
The building data is stored under `udx/bldg` in the extracted directory.

Open the `gml` file in a text editor and confirm the survey year, as in `<uro:surveyYear>2017</uro:surveyYear>` .

* This is necessary to check if the existing OSM data is mapped newer than the survey year, or if the PLATEAU data is older than the latest satellite image/aerial photograph.

## 2. Download the conversion script
Download [yuuhayashi / citygml-osm](https://github.com/yuuhayashi/citygml-osm) .

* The compiled script is distributed under the release distribution.
* The version of the conversion script will be updated, and it is recommended to use the latest version available at that time.

Extract the downloaded file and store `citygml-osm-osm-jar-with-dependencies.jar` in the file where the Plateau building data (in `citygml` format, under `udx/bldg` by default) is extracted.

## 3. Run scripts from 1st to 3rd
Start a Terminal (or Command Line, Powershell, etc.) and navigate to the folder where `citygml` was extracted.

Once there, type the following command.

```
$ java -Dfile.encoding=utf-8 -jar citygml-osm-jar-with-dependencies.jar 1st
```
```
$ java -Dfile.encoding=utf-8 -jar citygml-osm-jar-with-dependencies.jar 2nd
```
```
$ java -Dfile.encoding=utf-8 -jar citygml-osm-jar-with-dependencies.jar 3rd
```

In each command, the following processes are performed.

* 1st: Convert CityGML data to `.osm` format and save as `.osm` format.
* 2nd: Download the data in the data range of each `.osm` file from osm.org and save it as `.org.osm` format.
* 3rd: Compare and merge the `.osm` and `.org.osm` files to output and save the `mrg.osm` format file.

This process will result in four files with the same name but different file extensions ( `.gml,` `.osm,` `.org.osm` , and `.mrg.osm` ).
The above commands will produce the same results as if you had kicked in `run.sh` (or `run.bat` ), which is included in the distributed scripts.

The script will run against all `gml` files stored in the same directory. Therefore, if the number of files is large, it will take some processing time. If necessary, please separate directories accordingly.

## 4. Open `.mrg.osm` file with JOSM
Load a file with the extension `.mrg.osm` into JOSM.
After importing the `.mrg.osm` file in JOSM, display the OSM image layer in the background for clarity.

The goal of this process is to edit the imported file and remove any objects that you do not want to import.

## 5. Install the todo plugin
Follow these steps to enable the todo plugin.
* Open the JOSM menu and go to the Plug-in Settings tab.
* Check the todo plugin and press "OK" to install the plugin.
* After installation, restart JOSM to activate the installation.
* After restarting, activate the todo plugin from the list of plugins on the left side of the JOSM screen.
* Once the plugin is activated, the todo plugin will appear in the left pane of the JOSM screen.

## 6. Filter settings
Set the following four filters in the JOSM filters.

* `"MLIT_PLATEAU:fixme"=*`
* `"MLIT_PLATEAU:fixme"="PLATEAUデータで更新されています"` (means "Updated with PLATEAU data")
* `"MLIT_PLATEAU:fixme"="delete 削除されます"` (means ""delete" that will be deleted")
* `type:node`

In the composite file `.mrg.osm` , if there is an overlap between existing OSM data and PLATEAU data in the same area, the target object is given the tag `"MLIT_PLATEAU:fixme"` . A human must check if this object should be imported and handle it appropriately.

Basically, by processing the `"MLIT_PLATEAU:fixme"=*` , you can create a file that is ready for uploading.
Use the JOSM filter to make it easier to find the target object.

### Explanation of the values contained in `"MLIT_PLATEAU:fixme"=*`
* Either of the following two values is stored in `"MLIT_PLATEAU:fixme"=*` .
  * `"MLIT_PLATEAU:fixme"="PLATEAUデータで更新されています"`
  * `"MLIT_PLATEAU:fixme"="delete 削除されます"`

## 7 Edit `.mrg.osm`
Edit `.mrg.osm` in JOSM.
* **When editing this data, NEVER perform any operations other than "delete object" such as "move object" or "change tags".**
* The subsequent `update` process operates on the assumption that only unnecessary objects are deleted and that the tag contents have not been changed. If the object content is changed, the operation will be "operation indefinite".

### Checking data to which `"MLIT_PLATEAU:fixme"` is not assigned

Activate the filter `"MLIT_PLATEAU:fixme"=*` to display only untagged data (i.e., objects with PLATEAU data as is).

Set the background image to a satellite photo taken as recently as possible, such as Bing or Maxar, and delete any building objects that clearly do not exist. At this time, compare the survey year of the gml data confirmed in step **1.** with the year the background image was taken, and always delete objects according to the newer year.

### Correspondence to data with `"MLIT_PLATEAU:fixme"=*`
With the filters `type:node` and `"MLIT_PLATEAU:fixme"="delete 削除されます"` activated, inverse `"MLIT_PLATEAU:fixme"="delete 削除されます"` , press <Ctrl + A> and select all objects in the layer Select all objects in the layer.

This will ensure that no nodes are selected when selecting objects.

#### **Processing of `"MLIT_PLATEAU:fixme"="PLATEAUデータで更新されています"`**
* Objects that will be replaced by PLATEAU data geometry from existing OSM data.
* Visual checks and corrections are performed to ensure that the geometry is acceptable for uploading.

* These building shape data have the following characteristics

  * Building geometry: PLATEAU side geometry is used
  * Tags: both existing and PLATEAU merged (see GitHub for tag merging policy)

* Next, select the layer, activate the filters `type:node` and `"MLIT_PLATEAU:fixme" = "PLATEAUデータで更新されています"` , press <Ctrl + A> and select all objects in the layer.
* Once all tagged objects are selected, press the `+` button in the todo plugin to store the object in the todo list and create a checklist.

* Set two background images, a tile image from `osm.org` and a new satellite photo of the year taken, and check their position in relation to the existing buildings in OSM.
  * If you are satisfied that the object geometry and tags are ok and can be imported, press the `✔` button in the todo plugin.
  * If the object with the tag `"MLIT_PLATEAU:fixme"="PLATEAUデータで更新されています"` is in the wrong position, delete the object. By deleting it, the existing OSM data will take precedence and the geometry will not be updated.

* DO NOT delete the tag `"MLIT_PLATEAU:fixme"="PLATEAUデータで更新されています"` for use in the next process.

#### Processing for `"MLIT_PLATEAU:fixme"="delete 削除されます"`
* Objects with this tag are proposed to be deleted for some reason.
* For example, if an existing building with 3 units is to be merged into a PLATEAU object with 2 units, the one unit that was left out of the merger will be "Deleted".

* As with the `"PLATEAUデータで更新されています"` object, select the layer, activate the filters `type:node` and `"MLIT_PLATEAU:fixme"="delete 削除されます"` and press <Ctrl + A> to select all objects in the layer.
* Once all tagged objects are selected, press the `+` button in the todo plugin to store the target object in the todo list and create a checklist.

  * If you are satisfied that the object can be deleted by comparing it with the tile image on `osm.org` or a new satellite photo of the year it was taken, press the `✔` button in the todo plugin to delete it from the list and leave the object with this tag: `"MLIT_PLATEAU:fixme"="delete 削除されます"` .

  * If you want to keep this building shape, delete the **tag** `"MLIT_PLATEAU:fixme"="delete 削除されます"` .
  * In addition, there should often be a building shape in the immediate vicinity that is being replaced. Turn the filter On/Off to display the object and modify nearby building features accordingly. In some cases, it is important to copy tags such as `name` from nearby building objects to the building object you want to keep.
  * Delete all objects that overlap with the building object you want to keep so that you can keep the target building shape.

#### In case you are not sure what to do?
In most cases, PLATEAU building data is very superior and there is no need to worry about replacing the geometry.

However, PLATEAU data is not perfect either. If it is difficult to determine from aerial photos such as Bing what the current building conditions are and whether the building shape in the PLATEAU dataset is indeed correct, use the following clues to gather information.

* **Check the history of existing data**
  * Check the history of existing data and other information to determine what information was used to draw the shape.
  * PLATEAU data is mainly created from data acquired around 2016, and for building objects added later than that, the newly drawn buildings may be correct. Buildings whose geometry was drawn based on survey information after 2016 in particular may be more reliable.

* **Refer to the most recent aerial photograph.**
  * Even if a building object was added to OSM after 2016, the referenced aerial/satellite photo may be older than 2016.
  * The authenticity of the data can be confirmed by comparing it to aerial photographs taken newer than 2016.

* **Still, we just can't discern or make a decision.**
  * Please pause your work, add the name of the mesh in question, and consult with us via Slack channel or other means.

For areas where PLATEAU data has been imported, we plan to use the Tasking Manager to perform a post-check process for geometry, etc., after the work is complete. Any major mistakes can be checked during the post-verification process.

## 8. Save the checked file
Save the checked file as a file with a name.

Save the target layer from the JOSM menu or from the JOSM layer list.

When saving, name the file `checked.osm` .

## 9. Conversion to a file for upload (run 4th script)
Convert the `checked.osm` file to a file for uploading.

From the Terminal (or Command Line, Powershell, etc.), enter the following command.
```
$ java -Dfile.encoding=utf-8 -jar citygml-osm-jar-with-dependencies.jar 4th
```
This will generate `upload.osm` based on `checked.osm` .

## 10. Upload to `osm.org`
Open `upload.osm` in JOSM and upload to `osm.org` .

`Validation` is performed with `upload.osm` opened in the layer.

Review any warnings or errors that appear as a result of the validation, and then upload the data.
For example, there may be error handling of duplicate nodes at the edges, or relationships where no members exist.

When uploading, please add the following tags to the change set
```
 source = MLIT_PLATEAU
 source_ref = https://wiki.openstreetmap.org/wiki/MLIT_PLATEAU/imports_outline
```
The following format is also recommended as a comment on the change set
The mesh number is the first 8 digits of the file you are working on.
```
Plateau import: [target municipality] [mesh number].
```
Example: Plateau import: Niiza-shi 53395404

## 11. Edit the target list in OSM wiki
OSM wiki has a list of file names for each municipality to be imported.

* https://wiki.openstreetmap.org/wiki/JA:MLIT_PLATEAU/imports_list

For each mesh number that has been imported, please note that it has been imported.
If there are any remarks, please state them in the remarks field.

---
This is the end of the process.
